### ISSUE TITLE
STAR WARS™: X-Wing Alliance (361670)

### ISSUE THREAD

**Issue by bjoern-tantau** [ValveSoftware/Proton#6125]
# Compatibility Report
- Name of the game with compatibility issues: STAR WARS™: X-Wing Alliance 
- Steam AppID of the game: 361670

## System Information
Steam Deck 
- GPU: AMD Custom GPU 0405 (vangogh, LLVM 13.0.0, DRM 3.45, 5.13.0-valve21.1-1-neptune-02211-gc54cda5a36f3)
- Driver/LLVM version: 4.6 (Compatibility Profile) Mesa 22.0.2 (git-71648faf15) 
- Kernel version: 5.13.0-valve21.1-1-neptune-02211-gc54cda5a36f3
- Link to full system information report as [Gist](https://gist.github.com/bjoern-tantau/d39a868a16a97fa1258d5e781454b0f3):
- Proton version: 7.0-4

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 

Game gives the message:

ERROR: Joystick not detected! Game will not work properly without a joystick attached. 

## Reproduction

Select the standard controller configuration of Gamepad with camera controls, start the game, enter a pilot name and confirm.

**Comment by bjoern-tantau** [comment]
Proton log: https://gist.github.com/bjoern-tantau/dfd8da78227ba10a000235fd5cd79635

**Comment by bjoern-tantau** [comment]
I have also tried installing the X-Wing Alliance Upgrade Mega Patch from https://www.xwaupgrade.com/download/installers/XWAU2020.exe in the hopes that it would bring better compatibility, but I couldn't even run the installer: https://gist.github.com/bjoern-tantau/3bbd58309fd5851686428910b9d765c6

**Comment by alesliehughes** [comment]
Even on windows a joystick is a requirement.

**Comment by bjoern-tantau** [comment]
> Even on windows a joystick is a requirement.

But it's rated as playable for the Steam Deck. And technically the Deck comes with two joysticks which could be presented as such to the game.

**Comment by alesliehughes** [comment]
> > Even on windows a joystick is a requirement.
> 
> But it's rated as playable for the Steam Deck. And technically the Deck comes with two joysticks which could be presented as such to the game.

That should be true. I read the comments and thought it was being played on a PC not the steam deck.

**Comment by bjoern-tantau** [comment]
I think the powers that be have heard me and did a configuration update. I can now use the controller as a joystick. Now I just need a way to install XWAU, but that is probably beyond Valve.

**Comment by jrobio** [comment]
I'm having issues with joystick detection on my steam deck as well. By any chance, does your right thumbstick do anything? I imagine expected behavior is rudder control in the least when pressing the right thumb stick along the x axis. However, my right thumb stick currently has no input effect.

I own this game on both steam and gog. When running XWAUpgrade managed on lutris and through game mode, I had the odd behavior of the right trigger controlling roll. Spent some time googling around, utterly baffled.

**Comment by bjoern-tantau** [comment]
> I'm having issues with joystick detection on my steam deck as well. By any chance, does your right thumbstick do anything? I imagine expected behavior is rudder control in the least when pressing the right thumb stick along the x axis. However, my right thumb stick currently has no input effect.

Yes, that is the correct behaviour. Remember, the game is from 1999, where twin sticks didn't exist. You can roll by holding the B button and using the left joystick. This has been the standard behaviour since the first X-Wing game. XWAU added proper controller support.

Personally I set the right joystick to be the left one and I put the arrow keys and enter key on the left joystick. Don't know if I will keep it that way.

**Comment by jrobio** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/6125#issuecomment-1231533426

Gratifying to know I'm in the realm of expected behavior for this game. Hopefully, if I manage to get XWAU working on the steam version, steam input will be able to identify my right stick and apply the correct configuration. Thank you for the informative reply.

**Comment by Wedge009** [comment]
I never played this game before until today, I was only able to get it running on Windows. I tried stable Proton 9.0-4, 8.0-5, 7.0-6, 6.3-8 - in all cases the Proton system ceases after clicking 'Play X-Wing Alliance' on the launcher.

This report is for Proton 9.0-4, running on Ubuntu 24.04:
[steam-361670.log](https://github.com/user-attachments/files/19973758/steam-361670.log)

The complaint about pid is odd - as though the system isn't properly launching the main game?

I'm aware of the 'upgrade' project, I haven't touched any of that. This is running the game as installed through Steam.

**Comment by Wedge009** [comment]
This [GOG thread](https://www.gog.com/forum/star_wars_xwing_and_tie_fighter_series/solved_issue_xwing_alliance_wont_launch_on_windows_10) describes the same issue, and mentions DirectPlay. Could this be relevant for Proton? I'm not sure how to tell if DirectPlay is enabled for Proton or how to do so if it isn't.

**Comment by Wedge009** [comment]
No apparent change with Proton 10.0-3.

**Comment by Wedge009** [comment]
At least something seems to be launched with Proton 11, but I just get a black screen. I can hear audio.

At a guess, maybe my UHD display is 'too big' for an older game?
